homeassistant/components/plugwise/climate.py

```diff
@@ -1,141 +1,152 @@
-    HVAC_MODE_OFF,
     SUPPORT_PRESET_MODE,
     SUPPORT_TARGET_TEMPERATURE,
 )
-from homeassistant.const import (
-    ATTR_TEMPERATURE,
-    CONF_HOST,
-    CONF_NAME,
-    CONF_PASSWORD,
-    CONF_PORT,
-    CONF_USERNAME,
-    TEMP_CELSIUS,
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+
+from . import SmileGateway
+from .const import (
+    DEFAULT_MAX_TEMP,
+    DEFAULT_MIN_TEMP,
+    DOMAIN,
+    SCHEDULE_OFF,
+    SCHEDULE_ON,
+    THERMOSTAT_ICON,
 )
-from homeassistant.exceptions import PlatformNotReady
-import homeassistant.helpers.config_validation as cv
+
+HVAC_MODES_HEAT_ONLY = [HVAC_MODE_HEAT, HVAC_MODE_AUTO]
+HVAC_MODES_HEAT_COOL = [HVAC_MODE_HEAT_COOL, HVAC_MODE_AUTO]
 
 SUPPORT_FLAGS = SUPPORT_TARGET_TEMPERATURE | SUPPORT_PRESET_MODE
 
 _LOGGER = logging.getLogger(__name__)
 
-# Configuration directives
-CONF_MIN_TEMP = "min_temp"
-CONF_MAX_TEMP = "max_temp"
-CONF_LEGACY = "legacy_anna"
-
-# Default directives
-DEFAULT_NAME = "Plugwise Thermostat"
-DEFAULT_USERNAME = "smile"
-DEFAULT_TIMEOUT = 10
-DEFAULT_PORT = 80
-DEFAULT_ICON = "mdi:thermometer"
-DEFAULT_MIN_TEMP = 4
-DEFAULT_MAX_TEMP = 30
-
-# HVAC modes
-HVAC_MODES_1 = [HVAC_MODE_HEAT, HVAC_MODE_AUTO]
-HVAC_MODES_2 = [HVAC_MODE_HEAT_COOL, HVAC_MODE_AUTO]
-
-# Read platform configuration
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {
-        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
-        vol.Required(CONF_PASSWORD): cv.string,
-        vol.Required(CONF_HOST): cv.string,
-        vol.Optional(CONF_LEGACY, default=False): cv.boolean,
-        vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
-        vol.Optional(CONF_USERNAME, default=DEFAULT_USERNAME): cv.string,
-        vol.Optional(CONF_MIN_TEMP, default=DEFAULT_MIN_TEMP): cv.positive_int,
-        vol.Optional(CONF_MAX_TEMP, default=DEFAULT_MAX_TEMP): cv.positive_int,
-    }
-)
 
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Smile Thermostats from a config entry."""
+    api = hass.data[DOMAIN][config_entry.entry_id]["api"]
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]["coordinator"]
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
-    """Add the Plugwise (Anna) Thermostat."""
-    api = haanna.Haanna(
-        config[CONF_USERNAME],
-        config[CONF_PASSWORD],
-        config[CONF_HOST],
-        config[CONF_PORT],
-        config[CONF_LEGACY],
-    )
-    try:
-        api.ping_anna_thermostat()
-    except OSError:
-        _LOGGER.debug("Ping failed, retrying later", exc_info=True)
-        raise PlatformNotReady
-    devices = [
-        ThermostatDevice(
-            api, config[CONF_NAME], config[CONF_MIN_TEMP], config[CONF_MAX_TEMP]
-        )
+    entities = []
+    thermostat_classes = [
+        "thermostat",
+        "zone_thermostat",
+        "thermostatic_radiator_valve",
     ]
-    add_entities(devices, True)
+    all_entities = api.get_all_devices()
```

Comment:
I'd call this `all_devices` instead of `all_entities`. They are not entity instances.

Reply:
Addressed in the child-PR now.